scripts/globals/conquest.lua

```diff
@@ -85,10 +85,10 @@ local function hasOutpost(player, region)
     local region = region + 5
     local hasOP = player:hasTeleport(player:getNation(), region)
     if not hasOP then
-        if UNLOCK_OUTPOST_WARPS == 2 then
+        if UNLOCK_OUTPOST_WARPS == 1 then
```

Comment:
```
        if UNLOCK_OUTPOST_WARPS == 1 then
            hasOP = true
```
if setting is 1, the check for if you have access is always true. that is what this says. That is not what should happen here.

Reply:
I honestly can only say to try to use the current code or what I have posted. The current 1 doesn't allow Li'Telor, Kuzotz, Vollbow, Elshimo Lowlands, Elshimo Uplands. (Animation happens, no currency taken and no teleport) Why that is is maybe due to quest requirement besides the supply missions to be in those areas.

Reply:
this is what the code does. indisputably with even basic understanding of lua. The ONLY instance that should be always true is the value of 2 you changed it away from. You do not get access to tav+sky without:
setting=2, OR, actually have the unlock legit (and in one of those, there is no legit unlock)

Reply:
With option 1 Tu'lia and Tavnazia do not show up on the list at the NPC.

Reply:
Because the list is generated by `getAllowedTeleports`, not `hasOutpost`.

Reply:
So my question is I guess, why is there even a restriction on options 1 or 2 when it comes to `hasOutpost `? The list comes from `getAllowedTeleports` the last two teleports for option 2 won't be available anyways unless you choose that option. And the point of them is to have access to everything in those option 1 or 2 lists. I mean, it works like this.

Reply:
`hasOutpost` is used in other places to calculate fees and in the event finish to verify that that the packet was not altered and the players choice is indeed a valid choice from the menu.

Reply:
`        elseif UNLOCK_OUTPOST_WARPS == 1 then
            hasOP = region <= tpz.region.ELSHIMOUPLANDS`

I ment this, not fees and such, I understand that.

Reply:
The only outpost regions above Elshimo Uplands is Tu'Lia and Tavnazia, so if UNLOCK_OUTPOST_WARPS is 1 then it will return true for everything except those 2 regions, so if for some reason a player is able to select those options even though they're not in the menu, they will still be rejected.

Reply:
But it does not work for everything below, nothing above Quifim Island actually teleports you anywhere with option 1.

Reply:
Okay the problem seems to be the `local region = region +5`. I'm not sure if this +5 is needed for the `hasTeleport` call, but it is screwing up the region comparison if `UNLOCK_OUTPOST_WARPS == 1`.

Reply:
```lua
local function hasOutpost(player, region)
    local hasOP = player:hasTeleport(player:getNation(), region + 5)
    if not hasOP then
        if UNLOCK_OUTPOST_WARPS == 2 then
            hasOP = true
        elseif UNLOCK_OUTPOST_WARPS == 1 then
            hasOP = region <= tpz.region.ELSHIMOUPLANDS
        end
    end
    return hasOP
end
```
that probably fixes the problem without breaking anything, assuming everything was working prior to [this change](https://github.com/project-topaz/topaz/commit/6507e3d88db53d750858af4f2978c46c4fbef8f7).

Reply:
I will test it out right now. Thanks!

Reply:
Seems to be working good now! on 

> ```lua
> local function hasOutpost(player, region)
>     local hasOP = player:hasTeleport(player:getNation(), region + 5)
>     if not hasOP then
>         if UNLOCK_OUTPOST_WARPS == 2 then
>             hasOP = true
>         elseif UNLOCK_OUTPOST_WARPS == 1 then
>             hasOP = region <= tpz.region.ELSHIMOUPLANDS
>         end
>     end
>     return hasOP
> end
> ```
> 
> that probably fixes the problem without breaking anything, assuming everything was working prior to [this change](https://github.com/project-topaz/topaz/commit/6507e3d88db53d750858af4f2978c46c4fbef8f7).

This seems to be working @cocosolos ! Would you mind commiting this please? :)